### PR TITLE
Comparison notebook between deep learning and regression-based forecasts

### PR DESCRIPTION
**Pull request by geek-yang**
To highlight that with "s2spy", users can easily generate some skillful baseline forecasts for their deep learning study, we create this notebook, in which we compare forecasts with LSTM and ridge.

### CODE REVIEW

File: workflow/comp_pred_ridge_and_LSTM.ipynb [hunk 1]
```diff
@@ -0,0 +1,2196 @@
...
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "ground_truth = test_y_torch.squeeze().numpy()\n",
+    "\n",
+    "print(f\"The MSE loss is {hist_test[0]:.3f}\")\n",
```
Comment (BSchilperoort) [inline workflow/comp_pred_ridge_and_LSTM.ipynb:1953]: Can you add the MSE loss for the mean ground truth here as well? Should be simple to calculate.

File: workflow/comp_pred_ridge_and_LSTM.ipynb [hunk 2]
```diff
@@ -0,0 +1,2196 @@
...
...q6mqEh4ejvLxcV6Z///5ISUnBqVOnsG3bNoiiiPDwcGg0mnrrXbZsGVavXo01a9YgNzcXzs7OiIiIQGVlpSlui4iIiMxco7fCMIXLly9DLpcjOzsbQ4cONVjm2LFjePDBB3HmzBl07dq1znVRFOHt7Y3Zs2djzpw5AACVSgWFQoHU1FQ8++yz94yDW2GYN41Wg6NlR3G54jI6OHVAkDwIMhuZ1GEREZHEmvL53aRZYC1NpVIBADw9PQ1eLy8vR0pKCvz8/ODj42OwTFFREUpLSxEWFqY75+7ujkGDBmH//v0GE6CqqipUVVXpjtVq9f3cBrWg7cXbseTgElyquKQ7p3BSYF7wPIT5hjXwTCLrxS8NRHVJ2gV2J61Wi7i4OAwZMgSBgYF61z744AO4uLjAxcUF3333HTIzM2FnZ2ewntLSUgCAQqHQO69QKHTX7paUlAR3d3fdo77kiqS1vXg74rPi9ZIfACirKEN8Vjy2F2+XKDIi87W9eDsivozAlG1TkLAnAVO2TUHElxH890JWz2wSoJiYGJw4cQKff/55nWsTJkzADz/8gOzsbHTv3h3jx4836nie+fPnQ6VS6R7nz583Wt1kHBqtBksOLoGIuj22teeWHlwKjbb+sWFE1oZfGojqZxYJUGxsLLZu3Ypdu3ahU6dOda67u7ujW7duGDp0KL744gucPn0aaWlpButSKpUAgEuX9P/BX7p0SXftbvb29nBzc9N7kHk5Wna0zn/idxIhorSiFEfLjpowKiLzxS8NRA2TNAESRRGxsbFIS0vDzp074efn16jniKKoN2bnTn5+flAqldixY4funFqtRm5uLkJCQowWO5nW5YrLRi1H1NrxSwNRwyRNgGJiYrB+/Xps3LgRrq6uKC0tRWlpKX7//XcAwC+//IKkpCQcOXIE586dw759+/DMM8/A0dERTzzxhK6enj176lqEBEFAXFwcFi1ahPT0dBw/fhyTJk2Ct7c3oqKipLhNMoIOTh2MWo6oteOXBqKGSToLLDk5GQAQGhqqdz4lJQXR0dFwcHDAnj17sHLlSly7dg0KhQJDhw7Fvn379BZlzM/P180gA4C5c+eivLwcU6dOxfXr1/HII48gIyMDDg4OJrkvMr4geRAUTgqUVZQZbNIXIEDhpECQPEiC6IjMD780EDXMrNYBMhcttQ6QRiviYNFVlN2ohNz
```
Comment (BSchilperoort) [inline workflow/comp_pred_ridge_and_LSTM.ipynb:2154]: Again it would be nice to see the performance of the mean here. Technically it should be the mean of the training data for a more fair comparison.

**Review by BSchilperoort — APPROVED** [review]
Hi @geek-yang , nice work. I mostly have some textual comments and other small things. Feel free to merge it once you have resolved the issues :+1:

File: workflow/comp_pred_ridge_and_LSTM.ipynb [hunk 3]
```diff
@@ -0,0 +1,2196 @@
...
+    "# create preprocessor\n",
+    "preprocessor = preprocess.Preprocessor(\n",
+    "    rolling_window_size=25,\n",
+    "    detrend=\"linear\",\n",
+    "    subtract_climatology=True,\n",
+    ")\n",
+    "\n",
+    "# fit preprocessor with training data\n",
+    "preprocessor.fit(precursor_field.sel(time=slice(str(start_year),\n",
+    "                                                str(start_year + train_samples - 1))))"
```
[the remark this answers is not in the document]
Comment (geek-yang) [reply workflow/comp_pred_ridge_and_LSTM.ipynb:355]: Yep, this looks much better now:

```py
# fit preprocessor with training data
preprocessor.fit(
    precursor_field.sel(
        time=slice(str(start_year), str(start_year + train_samples - 1))
    )
)
```

File: workflow/comp_pred_ridge_and_LSTM.ipynb [hunk 4]
```diff
@@ -0,0 +1,2196 @@
...
+    "plt.plot(np.asarray(hist_train), 'b', label=\"Training loss\")\n",
+    "plt.plot(np.asarray(hist_valid), 'r', label=\"Validation loss\")\n",
+    "plt.xlabel('Epoch')\n",
+    "plt.ylabel('Loss')\n",
+    "plt.title(\"MSE\")\n",
+    "plt.legend()\n",
+    "\n",
+    "fig.add_subplot(1, 2, 2)\n",
+    "plt.plot(np.log(np.asarray(hist_train)), 'b', label=\"Training loss\")\n",
+    "plt.plot(np.log(np.asarray(hist_valid)), 'r', label=\"Validation loss\")\n",
```
[the remark this answers is not in the document]
Comment (geek-yang) [reply workflow/comp_pred_ridge_and_LSTM.ipynb:1874]: Good idea! Now using plt.semilogy instead of plt.plot 👍.

File: workflow/comp_pred_ridge_and_LSTM.ipynb (hunk 1 above)
Comment (geek-yang) [reply workflow/comp_pred_ridge_and_LSTM.ipynb:1953]: Yes, I will calculate MSE loss for the mean of training data (not ground truth, just to be fair).

File: workflow/comp_pred_ridge_and_LSTM.ipynb (hunk 2 above)
Comment (BSchilperoort) [reply workflow/comp_pred_ridge_and_LSTM.ipynb:2464]: I would think something goes wrong then. If the MSE is 97, that means that the non-squared error is 9.8 K. 

I would expect the mean of the training to be much more like the mean of the ground truth, as in this image:
image

The validation SST varies between 22 and 25 degrees C. With that small variance I would not expect to ever get a MSE of 97.

Comment (geek-yang) [reply workflow/comp_pred_ridge_and_LSTM.ipynb:2464]: 😂 I thought it was because of the trend. For a fair comparison, we took the mean of the data preceding the test period (about 30 years). And 30 years of trend causes the big error.

Anyway, it is better to use climatology rather than the mean. This is fixed in another PR #11 by Jannes. Thanks for checking it.
